content/kubermatic/main/installation/upgrading/upgrade-from-2.21-to-2.22/_index.en.md

```diff
@@ -0,0 +1,230 @@
+
++++
+title = "Upgrading from 2.21 to 2.22"
+date = 2023-02-13T00:00:00+01:00
+weight = 10
++++
+
+{{% notice note %}}
+Upgrading to KKP 2.22 is only supported from version 2.21. Do not attempt to upgrade from versions prior to that and apply the upgrade step by step over minor versions instead (e.g. from [2.20 to 2.21]({{< ref "../upgrade-from-2.20-to-2.21/" >}}) and then to 2.22). It is also strongly advised to be on the latest 2.21.x patch release before upgrading to 2.22.
+{{% /notice %}}
+
+This guide will walk you through upgrading Kubermatic Kubernetes Platform (KKP) to version 2.22. For the full list of changes in this release, please check out the [KKP changelog for v2.22](https://github.com/kubermatic/kubermatic/blob/main/docs/changelogs/CHANGELOG-2.22.md). Please read the full document before proceeding with the upgrade.
+
+## Pre-Upgrade Considerations
+
+KKP 2.22 adjusts the list of supported Kubernetes versions and drops a couple of Kubernetes releases without upstream support. Support for Kubernetes 1.23 and below has been removed. As such, all user clusters should be upgraded to Kubernetes 1.24. Existing Kubernetes 1.23 user clusters with containerd will not block the upgrade procedure, but they will be upgraded to 1.24 automatically.
+
+### Removal of Docker Support
+
+Support for Docker as container runtime has been removed due to removal of `dockershim` in upstream Kubernetes. The only supported
+container runtime in KKP 2.22 is therefore containerd. As such, the upgrade will fail if `kubermatic-installer` detects any user clusters
+with Docker as container runtime.
+
+It is necessary to migrate **existing clusters and cluster templates** to containerd before proceeding. This can be done either via the Kubermatic Dashboard
+or with `kubectl`. On the Dashboard, just edit the cluster or cluster template, change the _Container Runtime_ field to `containerd` and save your changes.
+
+![Change Container Runtime](/img/kubermatic/main/installation/upgrade_container_runtime.png?classes=shadow,border&height=200 "Change Container Runtime")
+
+If using `kubectl`, update the `containerRuntime` field that is part of the [`Cluster` spec]({{< ref "../../../references/crds/#clusterspec" >}})
+and replace `docker` with `containerd`, e.g. by using `kubectl edit cluster <cluster name>`:
+
+```yaml
+# before
+apiVersion: kubermatic.k8c.io/v1
+kind: Cluster
+metadata:
+  name: wq7dl64vbc
+  [...]
+spec:
+  containerRuntime: docker
+  [...]
+---
+# after
+apiVersion: kubermatic.k8c.io/v1
+kind: Cluster
+metadata:
+  name: wq7dl64vbc
+  [...]
+spec:
+  containerRuntime: containerd
+  [...]
+```
```

Comment:
This feels a bit overly handholdy. Can we just document

```
kubectl patch cluster CLUSTER_NAME --type=merge -p '{"spec":{"containerRuntime":"containerd"}}'
```

Reply:
Sure thing.

Reply:
Done!